pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -107,27 +107,31 @@ func (h *appHandler) apply(ctx context.Context, appRev *v1beta1.ApplicationRevis
 		UID:        h.app.UID,
 		Controller: pointer.BoolPtr(true),
 	}}
-	for _, comp := range comps {
-		comp.SetOwnerReferences(owners)
-		newComp := comp.DeepCopy()
-		// newComp will be updated and return the revision name instead of the component name
-		revisionName, err := h.createOrUpdateComponent(ctx, newComp)
-		if err != nil {
-			return err
-		}
-		// find the ACC that contains this component
-		for i := 0; i < len(ac.Spec.Components); i++ {
-			// update the AC using the component revision instead of component name
-			// we have to make AC immutable including the component it's pointing to
-			if ac.Spec.Components[i].ComponentName == newComp.Name {
-				ac.Spec.Components[i].RevisionName = revisionName
-				ac.Spec.Components[i].ComponentName = ""
+	isRevisionOnly := ac.Annotations[oam.AnnotationAppRevisionOnly] == "true"
+	// don't create components if revision-only annotation is set
+	if !isRevisionOnly {
```

Comment:
I think we can make a function for the revisionOnly case, so the code could more clear like:

```
if isRevisionOnly{
    revisionOnlyApply(..)
}

the reset code...
```

Reply:
Sure